parse/analyze.go

```diff
@@ -95,6 +96,11 @@ func (b *blockContext) variableExists(name string) bool {
 	return ok
 }
 
+// copyVariables copies both the user variables and anonymous variables.
+func (b *blockContext) copyVariables() (map[string]*types.DataType, map[string]map[string]*types.DataType) {
+	return maps.Clone(b.variables), maps.Clone(b.anonymousVariables)
```

Comment:
Do you need this to be a deep clone?  It's shallow, so in particular the map _inside_ the `anonymousVariables` map is not cloned.  I don't know the requirement

Reply:
We do not need to deep copy the nested map, because the nested map is representative of an object. Variable names have to be unique, so once an object is declared, its structure cannot change (e.g. it cannot get a new field).  I will add a note.